.github/workflows/backport.yml

```diff
@@ -16,19 +16,32 @@
 # Once a PR is "Squash and merge"'d and it has been correctly labeled
 # according to the naming convention.
 
+# This workflow file is copied straight from the given example:
+# https://github.com/tibdex/backport/blob/main/.github/workflows/backport.yml
+
 name: Backport
 on:
-  pull_request:
+  pull_request_target:
     types:
       - closed
       - labeled
 
 jobs:
   backport:
-    runs-on: ubuntu-18.04
     name: Backport
+    runs-on: ubuntu-latest
+    # Only react to merged PRs for security reasons.
+    # See https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request_target.
+    if: >
+      github.event.pull_request.merged
+      && (
+        github.event.action == 'closed'
+        || (
+          github.event.action == 'labeled'
+          && contains(github.event.label.name, 'backport')
```

Comment:
Is `github.event.pull_request.merged` not an event that happens when we merge?  So if we add a label after a merge this will never trigger since we would never pass this conditional.

For simplicity, we could just check here if the PR is merged (and closed?) and let the backport action handle the rest.  This is not currently checking all the logic as is since it would currently run (but not backport) if merged and closed but not labeled.

Reply:
`github.event.pull_request.merged` is an attribute on the PR which has triggered the event and not the event itself. The event follows the format `github.event.action` to tell you what the actual event triggered the workflow. I verified running some of my own tests and the `merged` attribute prints out true/false according to the merge state of the PR and not based off the trigger. Checking that `merged` is true seems like the correct thing here then

We then check 1) if merged 2) if the action was a PR being closed or a backport label being applied which seems correct to me. Does that make sense to you? The checking of the label is extra like you said just to save on running time since the action checks as well

Reply:
> `github.event.pull_request.merged` is an attribute on the PR which has triggered the event and not the event itself

This makes sense then!  Good to know.

As for the rest, we're already only triggering this action on `closed` or `labeled` per line 26/27 so this would be a bit simpler and catch all cases:
```
if: >
      github.event.pull_request.merged
      && contains(github.event.label.name, 'backport')
```

Reply:
That makes sense to me. I'm guessing the Actions author didn't know you could check like this

Reply:
I made the change to only trigger on label changes and not the PR closing. It makes it simple. You can only add the label to backport AFTER the PR is closed. That let's us know exactly what is being backported at the time we make the decision. That makes it so we don't assume we want to backport early and add the label but then only realize after a change grows that it may not be a good candidate to backport